build_tools/travis/flake8_diff.sh

```diff
@@ -137,12 +137,9 @@ check_files() {
 if [[ "$MODIFIED_FILES" == "no_match" ]]; then
     echo "No file outside sklearn/externals and doc/sphinxext/sphinx_gallery has been modified"
 else
-    # Default ignore PEP8 violations are from flake8 3.3.0
-    DEFAULT_IGNORED_PEP8=E121,E123,E126,E226,E24,E704,W503,W504
-    check_files "$(echo "$MODIFIED_FILES" | grep -v ^examples)" \
-           --ignore $DEFAULT_IGNORED_PEP8
+
+    check_files "$(echo "$MODIFIED_FILES" | grep -v ^examples)"
     # Examples are allowed to not have imports at top of file
-    check_files "$(echo "$MODIFIED_FILES" | grep ^examples)" \
-           --ignore $DEFAULT_IGNORED_PEP8 --ignore E402
+    check_files "$(echo "$MODIFIED_FILES" | grep ^examples)" --ignore E402
```

Comment:
We had a problem a while ago that `--ignore` replaces rather than adds on top of the default, this is why we have DEFAULT_IGNORED_PEP8 env variable. You can probably use git blame and look at the commit message or PR for more details. Can you please check the behaviour?

Reply:
Yeah I realized this would be the issue as I was going to bed last night. I will see if there is a way to avoid that issue.

Reply:
> We had a problem a while ago that --ignore replaces rather than adds on top of the default, this is why we have DEFAULT_IGNORED_PEP8 env variable. You can probably use git blame and look at the commit message or PR for more details. Can you please check the behaviour?

So I checked and this matches my expectation: `--ignore` flag overrides whatever is in setup.cfg, so you need to duplicate what is in setup.cfg if you want to add an additional warning to ignore. All in all I am not sure what to do with this issue/PR ...

If we had at least one non-default flake8 warning, it would be worth, to have a flake8 section in setup.cfg but it is not even the case ...

Reply:
After looking into it a bit more, I think it would make the most sense to move a flake8 config file into the examples directory with the --ignore flags for that directory and just use the default --ignore flags for the rest of the project.

Reply:
Can you actually have a per-folder flake8 config file? I would not think so from just looking at the flake8 doc.

Reply:
afaik when flake8 checks a file like ```/example_proj/examples/example.py``` it will check for config info in ```/example_proj/examples``` then in ```/example_proj``` with the info further up in the directory structure taking priority. so if you don't have a config file in ```/example_proj``` and you have one in ```/example_proj/examples``` the file in ```/example_proj/examples``` will take precedence.

Reply:
I was definitely not aware of this, nice! The problem is that the per-folder config file does not seem to be taken into account by `flake8 --diff` :disappointed:

Reply:
Ah the --diff thing is true. Hmm. This seems like it is probably not going to work.